feat: 도메인 엔티티 추가

yujongyeop opened the pull request:
[개요]
- ERD 설계도에 나온 엔티티들을 추가한다.

[작업 사항]
- 사용자
- 상품
- 주문
- 배송

yujongyeop committed:
feat: 사용자 엔티티 추가

[개요]
- 사용자 도메인 관련 엔티티를 추가한다.

[변경 사항]
- Member 엔티티 추가
- 사용자 권한 관련 MemberRole Enum 추가

yujongyeop committed:
feat: 상품 엔티티 추가

[개요]
- 상품 도메인 관련 엔티티를 추가한다.

[변경 사항]
- Product, ProductVariant, OptionType, OptionValue, Category 엔티티 추가
- productVariant와 ProductOption 매핑 테이블 추가

yujongyeop committed:
feat: 주문 엔티티 추가

[개요]
- 주문 도메인 관련 엔티티를 추가한다.

[변경 사항]
- Order, OrderDetail 엔티티 추가
- 주문 상태 관리 OrderStatus Enum 추가

yujongyeop committed:
refactor: 사용자 엔티티 수정

[개요]
- 사용자 엔티티의 필드의 접근 제한자를 private로 설정한다.
- 마지막 수정일을 어노테이션 기반으로 관리한다.

[변경 사항]
- 모든 필드에 private 접근 제한자 추가
- 마지막 수정일 필드에 @LastModifiedDate 어노테이션 추가

yujongyeop committed:
feat: 배송 엔티티 추가

[개요]
- 배송 도메인 관련 엔티티를 추가한다.

[변경 사항]
- Delivery 엔티티 추가
- 배송 상태 관리 DeliveryStatus Enum 추가

yujongyeop committed:
feat: 주소록 엔티티 추가

[개요]
- 사용자 도메인에 필요한 주소록 엔티티를 추가한다.

[변경 사항]
- Address 엔티티 추가

yujongyeop committed:
feat: 상품 엔티티 추가

[개요]
- 상품 리뷰 엔티티를 추가한다.

[변경 사항]
- ProductReview 엔티티 추가

yujongyeop committed:
feat: JpaAuditing 활성화

[개요]
- 엔티티 변경 감지를 위해 JpaAuditing을 활성화 시킨다.

[변경 사항]
- @EnableAuditing 어노테이션 추가

yujongyeop commented:
ERD 기반으로 엔티티를 작성 했습니다.
현재 일부 엔티티(쇼핑 카트)는 누락된 상태인데 해당 엔티티는 여러분이 ERD와 위 소스 코드를 참고하여 추가해주시기 바랍니다.
https://www.erdcloud.com/d/hmbPsDMrvgTosyJrM

LeeYuJin3613 commented on src/main/java/com/cju/shoppingmall/member/entity/Member.java:
```diff
@@ -0,0 +1,39 @@
+    @Column(nullable = false)
+    private MemberRole role;
+    @CreatedDate
+    private LocalDateTime createdAt;
```
NOT NULL 조건이 빠진 거 같아요!

yujongyeop replied:
명시적으로 NOT NULL을 사용하는 것이 좋긴합니다.
@CreatedDate 어노테이션은 JPA가 엔티티를 DB에 저장할 때 최초 생성된 시간을 입력해주는 어노테이션입니다.
따라서 JPA를 통해 저장되는 엔티티들은 모두 생성 시간이 자동으로 들어가게 됩니다.
즉, JPA 상에서 어느정도 NULL을 허용하지 않습니다.(다만, NOT NULL 보장한다는 아님)
@LastModifiedDate도 비슷한 맥락입니다.

좋은 지적 감사합니다.😊

LeeYuJin3613 commented on src/main/java/com/cju/shoppingmall/product/entity/Product.java:
```diff
@@ -0,0 +1,48 @@
+    @Column(nullable = false)
+    private String description;
+    @Column(nullable = false)
+    private String thumbnail;
```
ERD에서 NULL이라고 되어있음

yujongyeop replied:
수정하겠습니다.

LeeYuJin3613 commented on src/main/java/com/cju/shoppingmall/product/entity/Product.java:
```diff
@@ -0,0 +1,48 @@
+    @Column(nullable = false)
+    private Long basePrice;
+    @Column(nullable = false)
+    private Double discountRate;
```
ERD에서 NULL이라고 되어있음

yujongyeop replied:
수정하겠습니다.

LeeYuJin3613 commented on src/main/java/com/cju/shoppingmall/product/entity/Product.java:
```diff
@@ -0,0 +1,48 @@
+    @ManyToOne(fetch = FetchType.LAZY)
+    private Category category;
+    @CreatedDate
+    private LocalDateTime createdAt;
```
ERD에서 NOT NULL이라고 되어있음

LeeYuJin3613 commented on src/main/java/com/cju/shoppingmall/product/entity/OptionType.java:
```diff
@@ -0,0 +1,20 @@
+    @Column(nullable = false)
+    private String name;
+    @Column(nullable = false)
+    private String displayName;
```
ERD에 NULL로 되어있음

yujongyeop replied:
이것은 ERD에 잘못 표기한 문제입니다.
ERD를 수정하겠습니다

LeeYuJin3613 commented on src/main/java/com/cju/shoppingmall/product/entity/OptionValue.java:
```diff
@@ -0,0 +1,22 @@
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    @ManyToOne(fetch = FetchType.LAZY)
+    private OptionType optionType;
```
nullable = false 추가 해야되는 거 아닌가요..? ->ManyToOne가 있어서 잘 모르겠음. 일단 적어놓을게요

yujongyeop replied:
반영 했습니다.
ManyToOne을 사용하면 Column을 사용할 수 없고, JoinColumn을 사용해야합니다!

LeeYuJin3613 commented on src/main/java/com/cju/shoppingmall/product/entity/ProductVariant.java:
```diff
@@ -0,0 +1,47 @@
+    private Boolean isActive;
+    private String fingerprint;
+    @CreatedDate
+    private LocalDateTime createdAt;
```
NOT NULL 조건이 빠져있음

yujongyeop committed:
feat: NOT NULL 제약 추가

[개요]
- ERD와 NOT NULL 제약 조건이 일치하지 않은 부분을 수정합니다.

[변경 사항]
- ERD를 참고하여 Member, OptionValue, Product, ProductVariant의 NOT NULL 제약 조건을 변경

Junho123a commented on src/main/java/com/cju/shoppingmall/delivery/entity/Delivery.java:
```diff
@@ -0,0 +1,40 @@
+    private String receiverName;
+    @Column(nullable = false)
+    private String receiverPhone;
+    @Column(nullable = false)
```
https://github.com/column(nullable = false)
private String zipCode; 요거 erd랑 안맞는듯 싶습니다

Junho123a committed:
장바구니

yujongyeop commented on src/main/java/com/cju/shoppingmall/cart/entity/Cart.java:
```diff
@@ -0,0 +1,24 @@
+import lombok.Getter;
+
+@Entity
+@Table(name = "cart")
```
P4. Cart에는 Table명을 명시적으로 쓰고, CartDetail에는 명시하지 않은 이유가 있으실까요?

Junho123a replied:
Order 참고해서 작성했는데 거기에도 order엔 테이블잇고 orderDetail엔 없길래 따라 적어봤습니다.
